hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/GenesisRecordsConsensusHookImpl.java

```diff
@@ -0,0 +1,150 @@
+
+        if (!systemAccounts.isEmpty()) {
+            createAccountRecordBuilders(systemAccounts, context, SYSTEM_ACCOUNT_CREATION_MEMO);
+            systemAccounts = Collections.emptyMap();
+        }
+
+        if (!stakingAccounts.isEmpty()) {
+            final var implicitAutoRenewPeriod = FUNDING_ACCOUNT_EXPIRY - consensusTime.getEpochSecond();
+            createAccountRecordBuilders(stakingAccounts, context, STAKING_MEMO, implicitAutoRenewPeriod);
+            stakingAccounts = Collections.emptyMap();
+        }
+
+        if (!miscAccounts.isEmpty()) {
+            createAccountRecordBuilders(miscAccounts, context, null);
+            miscAccounts = Collections.emptyMap();
+        }
+
+        if (!treasuryClones.isEmpty()) {
+            createAccountRecordBuilders(treasuryClones, context, TREASURY_CLONE_MEMO);
+            treasuryClones = Collections.emptyMap();
+        }
+    }
+
+    @Override
+    public void systemAccounts(@NonNull final Map<Account, CryptoCreateTransactionBody.Builder> accounts) {
+        systemAccounts.putAll(requireNonNull(accounts));
+    }
+
+    @Override
+    public void stakingAccounts(@NonNull final Map<Account, CryptoCreateTransactionBody.Builder> accounts) {
+        stakingAccounts.putAll(requireNonNull(accounts));
+    }
+
+    @Override
+    public void miscAccounts(@NonNull final Map<Account, CryptoCreateTransactionBody.Builder> accounts) {
+        miscAccounts.putAll(requireNonNull(accounts));
+    }
+
+    @Override
+    public void treasuryClones(@NonNull final Map<Account, CryptoCreateTransactionBody.Builder> accounts) {
+        treasuryClones.putAll(requireNonNull(accounts));
+    }
+
+    @VisibleForTesting
+    void setLastConsensusTime(@Nullable final Instant lastConsensusTime) {
+        consensusTimeOfLastHandledTxn = lastConsensusTime;
+    }
+
+    private void createAccountRecordBuilders(
+            @NonNull final Map<Account, CryptoCreateTransactionBody.Builder> map,
+            @NonNull final HandleContext context,
+            @Nullable final String recordMemo) {
+        createAccountRecordBuilders(map, context, recordMemo, null);
+    }
+
+    private void createAccountRecordBuilders(
+            @NonNull final Map<Account, CryptoCreateTransactionBody.Builder> map,
+            @NonNull final HandleContext context,
+            @Nullable final String recordMemo,
+            @Nullable final Long overrideAutoRenewPeriod) {
+        for (Map.Entry<Account, CryptoCreateTransactionBody.Builder> entry : map.entrySet()) {
+            final var recordBuilder = context.addPrecedingChildRecordBuilder(GenesisAccountRecordBuilder.class);
+
+            final var accountId = requireNonNull(entry.getKey().accountId());
+            recordBuilder.accountID(accountId);
+            if (recordMemo != null) recordBuilder.memo(recordMemo);
+
+            var txnBody = entry.getValue();
+            if (overrideAutoRenewPeriod != null)
```

Comment:
I think we should avoid if-statements without brackets (as a minimum for long lines).